### PR TITLE
manifest: nrf_modem v2.2.0

### PR DESCRIPTION
**Pull request by lemrey**
Updates to nrf_modem v2.2.0.

### CODE REVIEW

File: lib/nrf_modem_lib/Kconfig.modemlib [hunk 1]
```diff
@@ -189,6 +189,22 @@ config NRF_MODEM_LIB_MEM_DIAG_DUMP_PERIOD_MS
...
 
+menu "Interrupt priorities"
+
+DT_IPC := $(dt_nodelabel_path,ipc)
+
+config NRF_MODEM_IPC_IRQ_PRIO_DTS_OVERRIDE
+	bool "Override IPC IRQ in Device Tree"
+	default n
+	help
+	Override the IPC IRQ priority set in device tree
```
Comment (rlubos) [inline lib/nrf_modem_lib/Kconfig.modemlib:200]:
```suggestion
	  Override the IPC IRQ priority set in device tree
```

File: lib/nrf_modem_lib/Kconfig.modemlib [hunk 2]
```diff
@@ -189,6 +189,22 @@ config NRF_MODEM_LIB_MEM_DIAG_DUMP_PERIOD_MS
 
 endif
 
+menu "Interrupt priorities"
+
+DT_IPC := $(dt_nodelabel_path,ipc)
+
+config NRF_MODEM_IPC_IRQ_PRIO_DTS_OVERRIDE
+	bool "Override IPC IRQ in Device Tree"
+	default n
```
Comment (rlubos) [inline lib/nrf_modem_lib/Kconfig.modemlib:198]: No need for `default n`

Comment (lemrey) [reply lib/nrf_modem_lib/Kconfig.modemlib]: Thanks

File: lib/nrf_modem_lib/Kconfig.modemlib (hunk 1 above)
Comment (lemrey) [reply lib/nrf_modem_lib/Kconfig.modemlib]: Thanks

**Review by eivindj-nordic — APPROVED** [review]

File: lib/nrf_modem_lib/Kconfig.modemlib [hunk 3]
```diff
@@ -189,6 +189,20 @@ config NRF_MODEM_LIB_MEM_DIAG_DUMP_PERIOD_MS
 
 endif
 
+menu "Interrupt priorities"
+
+DT_IPC := $(dt_nodelabel_path,ipc)
+
+config NRF_MODEM_IPC_IRQ_PRIO_DTS_OVERRIDE
```
Comment (lemrey) [inline lib/nrf_modem_lib/Kconfig.modemlib:196]: @eivindj-nordic do you think these should have the `_LIB_` prefix?
They don't belong to libmodem itself.

**Review by lemrey — COMMENTED** [review]
I need a changelog entry for this.

File: lib/nrf_modem_lib/Kconfig.modemlib (hunk 3 above)
Comment (eivindj-nordic) [reply lib/nrf_modem_lib/Kconfig.modemlib:196]: Yes, they should!

**Comment by lemrey** [comment]
Added @divipillai to review changelog

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -363,8 +363,15 @@ Modem libraries
...
 * :ref:`nrf_modem_lib_readme` library:
 
+  * Added:
+
+    * The :kconfig:option:`CONFIG_NRF_MODEM_LIB_IPC_IRQ_PRIO_OVERRIDE` Kconfig option to override IPC IRQ priority from DeviceTree.
+    * The :kconfig:option:`CONFIG_NRF_MODEM_LIB_IPC_IRQ_PRIO` Kconfig option to configure the IPC IRQ priority when the Kconfig option :kconfig:option:`CONFIG_NRF_MODEM_LIB_IPC_IRQ_PRIO_OVERRIDE` is enabled.
+
   * Updated:
 
+    * The IPC IRQ priority is now set via DTS.
```
Comment (divipillai) [inline doc/nrf/releases/release-notes-changelog.rst:373]:
```suggestion
    * The IPC IRQ priority is now set using DTS.
```

**Review by divipillai — APPROVED** [review]

**Review by nordic-hani — APPROVED** [review]

**Comment by lemrey** [comment]
Updated with final SHA

**Review by rlubos — APPROVED** [review]

**Pull request merged**